### PR TITLE
Add support for displaying channel watermark/favicon on videos

### PR DESCRIPTION
**Pull request by syeopite**
watermark

Since this can be considered a user hostile feature, the icon is fully togglable within the preferences. 

Closes channel favicon component of #2395

### CODE REVIEW

File: src/invidious/config.cr [hunk 1]
```diff
@@ -44,6 +44,7 @@ struct ConfigPreferences
   property vr_mode : Bool = true
   property show_nick : Bool = true
   property save_player_pos : Bool = false
+  property show_channel_watermark : Bool = true
```
Comment (ChunkyProgrammer) [inline src/invidious/config.cr:47]:
```suggestion
  property show_channel_watermark : Bool = false
```

I believe this is supposed to be false?

File: config/config.example.yml [hunk 2]
```diff
@@ -823,6 +823,15 @@ default_user_preferences:
   ##
   #save_player_pos: false
 
+  ##
+  ## Show the channel's watermark on the bottom-right corner of videos
+  ##
+  ## Accepted values: true, false
+  ## Default: false
+  ##
+  #show_channel_watermark: true
```
Comment (ChunkyProgrammer) [inline config/config.example.yml:832]:
```suggestion
  #show_channel_watermark: false
```

This too, unless you wanted the default to be true?

Comment (syeopite) [reply config/config.example.yml:832]: Yeah those are intentional. I'd like for it to be enabled by default with an option to disable it for those who don't want it.

But if enough people want it be disabled by default, I'll change that.

Comment (ChunkyProgrammer) [reply config/config.example.yml:832]: @syeopite ah okay, i wasnt sure if it was intentional or not as it says `Default: false` on line 830 in thia file (which would need to change in that case)

Comment (syeopite) [reply config/config.example.yml:832]: Oops missed that. Thanks!

**Comment by SamantazFox** [comment]
I'm not really sure if that's a necessary "feature". As far as I know, it was never requested to us (I can't find a related issue) and it adds a lot of code, for something that might be left disabled.

As overlays goes, I'd prefer to see the end screen overlays (with a toggle in the preferences).

**Pull request closed without merging**